### ISSUE TITLE
BUG: inconsistent constructors for sparse and dense objects

### ISSUE THREAD

**Issue by wholmgren** [pydata/pandas#9272]
Continuing on my recent sparse kick...

I appreciate the fact that ``Series()``, ``DataFrame()``, and ``Panel()`` all return empty versions of themselves. I think it's reasonable to expect that their sparse counterparts would behave the same way, but this is only the case for ``SparseDataFrame()`` (it's even [tested](https://github.com/pydata/pandas/blob/master/pandas/sparse/tests/test_sparse.py#L865)). 

```python
pd.SparseSeries()
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-375-46c1881ac0a7> in <module>()
----> 1 pd.SparseSeries()

TypeError: __init__() missing 1 required positional argument: 'data'
```

and similar for ``SparsePanel()``.

I can submit a PR to change this behavior.

**Comment by jreback** [comment]
sure

**Comment by jreback** [comment]
closed by #9277
